### PR TITLE
[Minor] code refactors 

### PR DESCRIPTION
**Pull request by sendilkumarn**
**What kind of change does this PR introduce?**

Minor code refactoring.

> Wow the PR number is binary format of `13`, just in case if you are curious 😉 

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA

**Summary**

Few things will provide us minor performance wins and simply to get back in pace with the code base.

**Does this PR introduce a breaking change?**

NA

**Other information**

### CODE REVIEW

File: lib/utils/zero-config.js [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
         return 'development';
-    } else if (!optionsDevFlag && !optionsProdFlag && mode) {
+    } else if (mode) {
         return mode;
     }
     return 'production';
 }
 
-module.exports = function setDefaultConfigBasedOnEnvironment(webpackObject) {
+function getConfigurations(options, outputOptions) {
```
Comment (evenstensberg) [inline lib/utils/zero-config.js]: I think the old filename is more clear to what it is doing

File: lib/utils/compiler.js [hunk 2]
```diff
@@ -186,13 +186,12 @@ async function webpackInstance(opts) {
         throw new Error(processingErrors);
     }
 
-    const compiler = getCompiler(opts);
+    const compiler = getCompiler(options);
     let lastHash = null;
 
-    const ProgressPlugin = webpack.ProgressPlugin;
```
Comment (evenstensberg) [inline lib/utils/compiler.js]: Reason for doing so is mutation on several runs, I think it's fine, but if a regression comes up, this is the reason. We have to find an exact match, not instance match

File: lib/utils/group-helper.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ class GroupHelper {
         if (!arr) {
             return;
         }
-        var result = {};
+        const result = {};
```
Comment (evenstensberg) [inline lib/utils/group-helper.js]: result is mutated

File: lib/utils/zero-config.js (hunk 1 above)
Comment (sendilkumarn) [reply lib/utils/zero-config.js]: The old name is still there, this is just stubbed out function.

Comment (evenstensberg) [reply lib/utils/zero-config.js]: Noticed, thx for ensuring!

File: lib/utils/group-helper.js (hunk 3 above)
Comment (sendilkumarn) [reply lib/utils/group-helper.js]: This prevents the re-assignment and I think we will need this.

Comment (ematipico) [reply lib/utils/group-helper.js]: We can add the eslint rule `prefer-const` to be 100% sure (if it's not there)

Comment (evenstensberg) [reply lib/utils/group-helper.js]: Hmm, I'd say adding a property to a declared object is mutation but I'm probably wrong. Small thing though, nothing blocking

Comment (ematipico) [reply lib/utils/group-helper.js]: Well this is JavaScript, we don't have immutable structures. In JavaScript what it counts it's just the reassignment and if the object is not reassigned, `const` is the way to go.
But you're right. But in JS world that distinction doesn't exist :D

File: lib/utils/zero-config.js [hunk 4]
```diff
@@ -1,40 +1,38 @@
 function getEnvFromOptionsAndMode(mode, optionsObject) {
-    const optionsDevFlag = optionsObject.dev;
-    const optionsProdFlag = optionsObject.prod;
+    const { dev, prod } = optionsObject;
 
     if (process.env.NODE_ENV && process.env.NODE_ENV === 'production' || process.env.NODE_ENV === 'development') {
```
Comment (jamesgeorge007) [inline lib/utils/zero-config.js]:
```js
if (process.env.NODE_ENV && ['production', 'development'].includes(process.env.NODE_ENV)) {
```
Maybe :thinking:

Comment (sendilkumarn) [reply lib/utils/zero-config.js]: I assume, includes will be slower here. (eventhough it is very negligible but still)

Comment (ematipico) [reply lib/utils/zero-config.js]: create a new array just for a comparison? Not needed I think

**Pull request merged**